package/network/services/wireguard/files/wireguard_watchdog

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright (C) 2018 Aleksandr V. Piskunov <aleksandr.v.piskunov@gmail.com>.
+# Copyright (C) 2015-2018 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
+#
+# This watchdog script tries to reresolve and reconnect to inactive wireguard peers.
+# Use it for peers with a frequently changing dynamic IP. 
+# persistent_keepalive must be set, recommended value is 25 seconds.
+#
+# Run this script from cron every minute:
+# echo '* * * * * /usr/bin/wireguard_watchdog' >> /etc/crontabs/root
+
+
+. /lib/functions.sh
+
+check_peer_activity() {
+  local cfg=$1
+  local iface=$2
+  local public_key
+  local endpoint_host
+  local endpoint_port
+  local persistent_keepalive
+  local last_handshake
+  local idle_seconds
+
+  config_get public_key "${cfg}" "public_key"
+  config_get endpoint_host "${cfg}" "endpoint_host"
+  config_get endpoint_port "${cfg}" "endpoint_port"
+  config_get persistent_keepalive "${cfg}" "persistent_keepalive"
```

Comment:
Rather than requesting this from the configuration, it might be best to see if it's _currently_ set, via `wg show .. persistent-keepalives`.

Reply:
Done, persistent_keepalive is now being queried using 'wg'.
Check is looking for 'off' or empty values.

Reply:
if endpoint_port is empty, it should be the default value of 51820.

```shell
root@R2S:~# /usr/bin/wireguard_watchdog
Unable to find port of endpoint: `www.example.com:'
```

Reply:
and if the re-resolve result is the same as the current, is it okay to skip the `wg set ...` process and silent the log?
If the re-resolve result is correct, but the peer is offline for some reason... in this situation, I don't know if `wg set ...` over and over again will have some side-effect?